tests/scripts/all.sh

```diff
@@ -644,6 +647,28 @@ pre_check_tools () {
     "$@" scripts/output_env.sh
 }
 
+# Set up an include directory for baremetal tests. Exclude headers that the
+# configuration says we shouldn't rely upon.
+pre_setup_baremetal_usr_include () {
+    msg "info: setting up include directory for bare metal tests"
+    case " $RUN_COMPONENTS " in
+        *baremetal*)
+            cp -r /usr/include baremetal-usr-include
```

Comment:
This makes the test mechanism Unix-only.

Reply:
Unix-only isn't a problem in `all.sh`: this script is only meant to run on a CI system that we have some control on. But copying `/usr/include` does feel fragile. It might not be the correct directory to copy when cross-compiling, and for baremetal tests we're likely to want to cross-compile easily.

Reply:
New approach does not suffer from this issue.